tests/framework/extensions/clusters/aks/upgrades/k8s_upgrade.go

```diff
@@ -0,0 +1,79 @@
+package upgrades
+
+import (
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	aks "github.com/rancher/rancher/tests/framework/extensions/clusters/aks"
+	"github.com/sirupsen/logrus"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+// UpgradeK8SVersion is a helper function that upgrades the AKS cluster's k8s version
+func UpgradeK8SVersion(client *rancher.Client, cluster *management.Cluster, displayName, cloudCredentialID string) (*management.Cluster, error) {
+	logrus.Infof("======================================")
+	logrus.Infof("TEST CASE: Upgrading K8s Version")
+	logrus.Infof("======================================")
+
+	version_1_24_6 := "1.24.6"
```

Comment:
similar thoughts, i think pulling these values from config.yml would keep things more dynamic and flexible.

Reply:
The upgrading tests looks at the highest upgraded version of K8s available in Rancher and tests against that. So we could do that, but it would go against the design of the test.

Reply:
On second thought, I think I'll just scrap this upgrade test altogether. Since Ugur already has this changes in, I don't think it makes sense for it to have in multiple locations. Also avoids less work in terms of maintaining as well as you pointed out in an earlier comment @igomez06 .

Reply:
Yes, Markus, there should be now a versions package where you can get different version pools as a slice of strings. In the upgrade test, it takes that string of slices, if the config is not given it takes the latest available for the cluster, or the user still can declare the desired version to upgrade. I am open to talking about this as well, let me know if something seems odd